### DISCUSSION TITLE
More control over git commit message

### DISCUSSION THREAD

**Discussion by jalaziz** [GitbookIO/community#69] [Feature Requests]
### What's your feature/idea?

Allow more control over the git commit message.

There are three distinct issues here (ranked in order of importance):
1. Support customizable git commit prefixes. Today, GitBook commits are prefixed with `GitBook:`. However, this conflicts with the commit conventions in a repo where [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) are used. Allowing a customizable prefix would go a long way to improving commit message compliance.
2. Related to (1), support removing or modifying the GitBook change request number format from the git commit message. GitBook commits mention the change request number as `[#N]` which conflicts with Github's issue auto-linking. The auto-linking can cause confusion for anyone looking at the change in Github as it will link to an arbitrary issue or PR.
3. Support customizable git commit message bodies. It's always good to explain _why_ you're making the change. This is easy enough to do if the changes always originate from git, but not if they originate from GitBook.

### What's your desired outcome?

More control over the commit message format (particularly the subject line) when editing from GitBook with Github Sync enabled.

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by SamyPesse** [comment]
We just shipped this feature, you can now configure a custom template for commit messages. Combined with the new URLs for change-requests, you can also enable Autolink in GitHub to convert references like `GITBOOK-123` to change request URLs.

![image](https://user-images.githubusercontent.com/845425/217212240-76417f28-61ea-4a57-a0d7-df2bed5177b2.png)

Learn more in our documentation: https://docs.gitbook.com/getting-started/git-sync/commits

**Reply by jalaziz** [reply]
Amazing, thank you 🙏
